src/clarity-angular/datagrid/render/column-resizer.ts

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Directive, ElementRef, Renderer, OnDestroy, AfterViewInit, Output, EventEmitter} from "@angular/core";
+import {DomAdapter} from "./dom-adapter";
+import {DatagridRenderOrganizer} from "./render-organizer";
+
+@Directive({
+    selector: "clr-dg-column"
+})
+export class DatagridColumnResizer implements AfterViewInit, OnDestroy {
+
+
+    constructor(private el: ElementRef,
+                private renderer: Renderer,
+                private organizer: DatagridRenderOrganizer,
+                private domAdapter: DomAdapter) {
+
+        this.columnEl = el.nativeElement;
+    }
+
+    columnEl: Element;
+    columnRectWidth: number;
+    columnResizeBy: number = 0;
+
+    draggerEl: Element;
+    draggerElWidth: number;
+    draggerTracker: Element;
+
+    pageStartPositionX: number;
+    dragDistancePositionX: number; //relative to pageStartPosition
+
+    @Output("clrDgColumnResize") resizeEmitter: EventEmitter<any> = new EventEmitter();
+
+    dragWithinMinWidth: boolean = false;
+
+    private _columnMinWidth: number;
+
+    get columnMinWidth(): number {
+        return this._columnMinWidth;
+    }
+
+    set columnMinWidth(value: number) {
+        this._columnMinWidth = value || 96;
```

Comment:
This was really bad. 😞   Having `min-wdith: 96px` makes the code much simpler. 😃  I specified the min-width on `clr-dg-column` as `.datagrid-column` is also applied the selection column.